addons/binding/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/discovery/MulticastListener.java

```diff
@@ -0,0 +1,307 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.globalcache.discovery;
+
+import static org.openhab.binding.globalcache.GlobalCacheBindingConstants.*;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.util.Date;
+
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.openhab.binding.globalcache.internal.util.NetworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MulticastListener} class is responsible for listening for the GlobalCache device announcement
+ * beacons on the multicast address, and then extracting the data fields out of the received datagram.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+public class MulticastListener {
+    private final static Logger logger = LoggerFactory.getLogger(MulticastListener.class);
+
+    private MulticastSocket socket;
+
+    private String serialNumber = "";
+    private String vendor = "";
+    private String model = "";
+    private String softwareRevision = "";
+    private String hardwareRevision = "";
+
+    // GlobalCache-specific properties defined in this binding
+    private String uid = "";
+    private String ipAddress = "";
+    private String macAddress = "";
+
+    private Date lastUpdate;
+
+    // GC devices announce themselves on a multicast port
+    private static final String GC_MULTICAST_GROUP = "239.255.250.250";
+    private static final int GC_MULTICAST_PORT = 9131;
+
+    // Used for a basic soundness check on the received beacon.
+    // A beacon shorter than this is probably malformed.
+    private static final int GC_BEACON_MIN_LENGTH = 160;
+
+    // How long to wait in milliseconds for a discovery beacon
+    public static final int DEFAULT_SOCKET_TIMEOUT = 3000;
+
+    /*
+     * Constructor joins the multicast group, throws IOException on failure.
+     */
+    public MulticastListener() throws IOException, SocketException {
+        InetAddress ifAddress = NetworkUtil.getNetworkInterface();
```

Comment:
you might want to make this a configuration option in case a user has multiple interfaces available.